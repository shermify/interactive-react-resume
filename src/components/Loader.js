import React from 'react';
import PropTypes from 'prop-types';
import LoaderImage from '../images/icons/loader.svg';

const Loader = (props) => {
  const loaderStyle = {
    height: '312px',
    textAlign: 'center',
    paddingTop: '50px',
    display: props.visible ? 'block' : 'none',
  };

  return (
    <div style={loaderStyle}><LoaderImage /></div>
  );
};

Loader.propTypes = {
  visible: PropTypes.bool.isRequired,
};

export default Loader;
