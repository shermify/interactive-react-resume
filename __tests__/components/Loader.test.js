import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../../src/components/Loader';

describe('<Loader />', () => {
  it('style changes to display: none on visible = false', () => {
    const component = renderer.create(
      <Loader visible={false} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('style changes to display: block on visible = true', () => {
    const component = renderer.create(
      <Loader visible />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
