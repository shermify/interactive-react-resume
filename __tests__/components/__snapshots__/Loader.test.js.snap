// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> style changes to display: block on visible = true 1`] = `
<div
  style={
    Object {
      "display": "block",
      "height": "312px",
      "paddingTop": "50px",
      "textAlign": "center",
    }
  }
>
  <test-file-stub />
</div>
`;

exports[`<Loader /> style changes to display: none on visible = false 1`] = `
<div
  style={
    Object {
      "display": "none",
      "height": "312px",
      "paddingTop": "50px",
      "textAlign": "center",
    }
  }
>
  <test-file-stub />
</div>
`;
